### PR TITLE
Inject DR Credentials implementation

### PR DESCRIPTION
**Pull request by mpkrass7**
## Summary
Implementation created. Let me know if you need me to set up a unit test

### CODE REVIEW

File: src/datarobotx/idp/common/credentials_hooks.py [hunk 1]
```diff
@@ -25,27 +26,46 @@
...
-    @staticmethod
-    def validate_credentials(conf_creds: Dict[str, Any]) -> None:
+    def _validate_or_replace_credentials(self, conf_creds: Dict[str, Any]) -> None:
         """Confirm credentials appear to be valid and functional."""
         import datarobot as dr
 
         dr_creds = conf_creds["datarobot"]
+        try:
+            dr.Client(token=dr_creds["api_token"], endpoint=dr_creds["endpoint"])  # type: ignore
+        except ValueError:
```
Comment (brau0300) [inline src/datarobotx/idp/common/credentials_hooks.py:54]: does this catch all the cases correctly? e.g. key not in dr_creds, key is in dr creds but is not valid for auth, etc?

File: src/datarobotx/idp/common/credentials_hooks.py [hunk 2]
```diff
@@ -25,27 +26,46 @@
...
+            self._logger.warning(
+                "Could not authenticate with DataRobot. "
+                "Kedro will run dr.Client() for authentication."
+            )
+            client = dr.Client()
+            conf_creds["datarobot"]["api_token"] = client.token
+            conf_creds["datarobot"]["endpoint"] = client.endpoint
+            prediction_server_id = dr.PredictionServer.list()[0].id
 
-        client = dr.Client(token=dr_creds["api_token"], endpoint=dr_creds["endpoint"])  # type: ignore
```
Comment (brau0300) [inline src/datarobotx/idp/common/credentials_hooks.py:50]: don't we still need the test that the client works somewhere here?

**Review by brau0300 — CHANGES_REQUESTED** [review]

File: src/datarobotx/idp/common/credentials_hooks.py [hunk 3]
```diff
@@ -25,27 +26,46 @@
...
 
+    @property
+    def _logger(self):
+        return logging.getLogger(__name__)
+
     @hook_impl
     def after_catalog_created(
         self,
         catalog: DataCatalog,
-        conf_catalog: Dict[str, Any],
```
[the remark this answers is not in the document]
Comment (mpkrass7) [reply src/datarobotx/idp/common/credentials_hooks.py:32]: https://docs.kedro.org/en/stable/hooks/common_use_cases.html#common-use-cases

At least by kedro examples, the interface is commonly changed

File: src/datarobotx/idp/common/credentials_hooks.py (hunk 2 above)
Comment (mpkrass7) [reply src/datarobotx/idp/common/credentials_hooks.py:50]: I learned that `/version` is actually called when you run `dr.Client()` so no, the test is done implicitly by instantiating the client https://datarobot.slack.com/archives/C0E80BJS0/p1724853667451609

File: src/datarobotx/idp/common/credentials_hooks.py [hunk 4]
```diff
@@ -35,17 +40,52 @@ def after_catalog_created(
...
+            dr.Client(token=dr_creds["api_token"], endpoint=dr_creds["endpoint"])  # type: ignore
+        except ValueError:  # Invalid or unpopulated credentials
+            exception_occurred = True
+        except KeyError:  # No credentials.yml file
+            exception_occurred = True
+            conf_creds["datarobot"] = {
+                "prediction_environment_id": None,
+            }
+
+        finally:
```
Comment (mpkrass7) [inline src/datarobotx/idp/common/credentials_hooks.py:66]: Noting that I can delete this line, but I never get the chance to write `finally` after try/except

**Review by brau0300 — APPROVED** [review]

**Pull request merged**